sequencer.c

```diff
@@ -3162,6 +3162,25 @@ int setup_reflog_action(struct replay_opts *opts, const char *commit,
 	return 0;
 }
 
+int checkout_onto(struct replay_opts *opts,
+		  const char *onto_name, const char *onto,
+		  const char *orig_head, unsigned verbose)
+{
+	struct object_id oid;
+	const char *action = reflog_message(opts, "start", "checkout %s", onto_name);
+
+	if (get_oid(orig_head, &oid))
+		die(_("%s: not a valid OID"), orig_head);
+
+	if (checkout_base_commit(opts, onto, verbose, action)) {
+		apply_autostash(opts);
+		sequencer_remove_state(opts);
+		die(_("could not detach HEAD"));
+	}
+
+	return update_ref(NULL, "ORIG_HEAD", &oid, NULL, 0, UPDATE_REFS_DIE_ON_ERR);
```

Comment:
Is there a reason why UPDATE_REFS_DIE_ON_ERR is used? I am not sure the original would have died if the ref update failed.

Reply:
It’s used in update-ref.c here: https://github.com/git/git/blob/master/builtin/update-ref.c#L440-L442

Reply:
Yeah, I think you are right.

Reply:
I think in this case, we do not want to `die()`, though. We're in a library function that is supposed to be called from wherever, not just from code paths that want to die with a potentially confusing error message. `UPDATE_REFS_MSG_ON_ERR` would appear more appropriate in this case, to me. Looking at the [`update_refs()` code](https://github.com/git/git/blob/68372c88794aba15f853542008cda39def768372/refs.c#L1105-L1120), I am even more certain of this.

Reply:
Right. I see `update_refs()` is also used elsewhere in sequencer.c, and they all use `UPDATE_REFS_MSG_ON_ERR`.